olive/passes/onnx/inc_quantization.py

```diff
@@ -201,9 +257,140 @@ def _default_config(accelerator_spec: AcceleratorSpec) -> Dict[str, PassConfigPa
+            olive_model = model_proto_to_olive_model(
+                model,
+                tmp_model_path,
+                {
+                    "save_as_external_data": False,
+                    "all_tensors_to_one_file": True,
+                    "external_data_name": None,
+                },
+            )
+
+            # create evaluator for model
+            evaluator = OliveEvaluatorFactory.create_evaluator_for_model(olive_model)
+
+            # evaluate model
+            result = evaluator.evaluate(
+                olive_model,
+                [accuracy_metric],
+                self._accelerator_spec.accelerator_type,
+                [self._accelerator_spec.execution_provider],
+            )
+            joint_key = joint_metric_key(accuracy_metric.name, sub_type.name)
+            return result[joint_key].value
+
+        return eval_func
+
+    def _set_accuracy_criterion(self, sub_type):
+        # set accuracy criterion for INC according to Olive accuracy metric goal
+        goal_type = sub_type.goal.type
+        goal_value = sub_type.goal.value
+        higher_is_better = sub_type.higher_is_better
+
+        if goal_type == "max-degradation":
+            tolerable_loss = goal_value
+            criterion = "absolute"
+        elif goal_type == "min-improvement":
+            tolerable_loss = -goal_value
+            criterion = "absolute"
+        elif goal_type == "percent-max-degradation":
+            tolerable_loss = goal_value / 100
+            criterion = "relative"
+        elif goal_type == "percent-min-improvement":
+            tolerable_loss = -goal_value / 100
+            criterion = "relative"
+        else:
+            assert False, (
+                "Accuracy metric goal type for Intel® Neural Compressor quantization only suuport "
+                "'max-degradation', 'min-improvement', 'percent-max-degradation' and 'percent-min-improvement'."
+            )
+
+        return higher_is_better, criterion, tolerable_loss
+
+    def _set_tuning_config(self, run_config):
+        # set criterion and eval func for INC
+        # INC quantization without accuracy aware tuning situation:
+        #  1. without evaluator
+        #  2. with evaluator, without accuracy metric
+        #  3. with evaluator, with accuracy metric, without goal
+        # INC quantization with accuracy aware tuning situation:
+        #  1. with evaluator, with accuracy metric, with goal
+        try:
+            from neural_compressor.config import AccuracyCriterion, TuningCriterion
+        except ImportError:
+            raise ImportError(
+                "Please install `olive-ai[inc]` or `neural-compressor` to use Intel® Neural Compressor quantization"
+            )
+
+        _evaluator_config = deepcopy(run_config)
+
+        accuracy_criterion = AccuracyCriterion()
+        tuning_criterion = TuningCriterion()
+        eval_func = None
+        accuracy_metric = None
+
+        if self._eval_config is not None:
```

Comment:
>_eval_config [](http://example.com/codeflow?start=16&length=12)

it seems the config only use metrics definition. @jambayk , @trajepl should we use `metrics` instead `evaluator` here to avoid confusion?